modules/utxo-lib/test/bitgo/psbt/fromHalfSigned.ts

```diff
@@ -0,0 +1,98 @@
+import * as assert from 'assert';
+import { ecc as eccLib, getNetworkName, Network, networks, taproot } from '../../../src';
+import {
+  isPlaceholderSignature,
+  outputScripts,
+  ParsedSignatureScriptTaprootScriptPath,
+  parseSignatureScript,
+  verifySignatureWithPublicKey,
+} from '../../../src/bitgo';
+import { getHalfSignedTransaction2Of3, getPrevOutputs } from '../../transaction_util';
+import { getDefaultWalletKeys } from '../../testutil';
+import { getInputUpdate } from '../../../src/bitgo/psbt/fromHalfSigned';
+import { createOutputScript2of3 } from '../../../src/bitgo/outputScripts';
+import { normDefault } from '../../testutil/normalize';
+type InputType = outputScripts.ScriptType2Of3 | 'p2shP2pk';
+function getInputScripts(): InputType[] {
+  return [...outputScripts.scriptTypes2Of3, 'p2shP2pk'];
+}
+
+function runTest(scriptType: InputType, network: Network) {
+  const walletKeys = getDefaultWalletKeys();
+  describe(`getInputUpdate scriptType=${scriptType}, network=${getNetworkName(network)}`, function () {
+    let prevOutputs;
+    let tx;
+    before('create transaction', function () {
+      prevOutputs = getPrevOutputs(scriptType, BigInt(1e8), walletKeys.triple);
+      tx = getHalfSignedTransaction2Of3(walletKeys.triple, walletKeys.user, walletKeys.bitgo, scriptType, network, {
+        prevOutputs,
+      });
+    });
+    it('matches expected value', function () {
+      const vin = 0;
+
+      const result = getInputUpdate(tx.clone() /* FIXME: make getInputUpdate non-destructive */, vin, prevOutputs[0]);
+      const parsed = parseSignatureScript(tx.ins[vin]);
```

Comment:
Should this be using the other code, or using the fixtures? A future exists where the ParseSignatureScript code is deprecated / moved to utxo-bin, right?

Reply:
Or we use `parseSignatureScript` in `getInputUpdate`? They almost do the same thing

Reply:
My reason for rewriting rather than using `parseSignatureScript` was to get away from dependence on code that's been removed from upstream (classify). I think that reason still stands. So we could merge the two similar implementations, doing away with the use of classify, or consider `parseSignatureScript` deprecated, and plan to maintain and augment just what we need in `fromHalfSigned`

Reply:
let's get rid of `classify` in `parseSignatureScript`

Reply:
I wanted to do this for a while now https://github.com/BitGo/BitGoJS/pull/3001